In [1]:
from main import DA,df2,results,accuracy

c:\Users\emili\Documents\Escuela\Modelos_de_Credito\Credit_Punctuation_Model\main.py:23: FutureWarning: The 'method' keyword in Series.replace is deprecated and will be removed in a future version.
  df2[x] = df2[x].replace('_',method='ffill')
c:\Users\emili\Documents\Escuela\Modelos_de_Credito\Credit_Punctuation_Model\main.py:24: FutureWarning: Series.fillna with 'method' is deprecated and will raise in a future version. Use obj.ffill() or obj.bfill() instead.
  df2[x] = df2[x].fillna(method='ffill')
c:\Users\emili\Documents\Escuela\Modelos_de_Credito\Credit_Punctuation_Model\functions.py:41: FutureWarning: Series.fillna with 'method' is deprecated and will raise in a future version. Use obj.ffill() or obj.bfill() instead.
  return data.fillna(method='ffill')


KeyError: 'Credit_MixNum_Credit_Inquiries'

In [ ]:
DA

Customer_ID          Name  Annual_Income  Monthly_Inhand_Salary  \
0       CUS_0xd40  AaronMaashoh      1911412.0            1824.843333   
1       CUS_0xd40  AaronMaashoh      1911412.0               0.000000   
2       CUS_0xd40  AaronMaashoh      1911412.0               0.000000   
3       CUS_0xd40  AaronMaashoh      1911412.0               0.000000   
4       CUS_0xd40  AaronMaashoh      1911412.0            1824.843333   
...           ...           ...            ...                    ...   
99995  CUS_0x942c         Nicks      3962899.0            3359.415833   
99996  CUS_0x942c         Nicks      3962899.0            3359.415833   
99997  CUS_0x942c         Nicks      3962899.0            3359.415833   
99998  CUS_0x942c         Nicks      3962899.0            3359.415833   
99999  CUS_0x942c         Nicks      3962899.0            3359.415833   

       Interest_Rate  Delay_from_due_date  Num_of_Delayed_Payment  \
0                3.0                  3.0                     7.0   
1                3.0                 -1.0                     0.0   
2                3.0                  3.0                     7.0   
3                3.0                  5.0                     4.0   
4                3.0                  6.0                     0.0   
...              ...                  ...                     ...   
99995            7.0                 23.0                     7.0   
99996            7.0                 18.0                     7.0   
99997         5729.0                 27.0                     6.0   
99998            7.0                 20.0                     0.0   
99999            7.0                 18.0                     6.0   

       Changed_Credit_Limit  Num_Credit_Inquiries Credit_Mix  \
0                    1127.0                   4.0              
1                    1127.0                   4.0       Good   
2                       0.0                   4.0       Good   
3                     627.0                   4.0       Good   
4                    1127.0                   4.0       Good   
...                     ...                   ...        ...   
99995                 115.0                   3.0              
99996                 115.0                   3.0              
99997                 115.0                   3.0       Good   
99998                 115.0                   3.0       Good   
99999                 115.0                   3.0       Good   

       Outstanding_Debt  Credit_Utilization_Ratio  Credit_History_Age  \
0               80998.0                 26.822620               265.0   
1               80998.0                 31.944960               265.0   
2               80998.0                 28.609352               267.0   
3               80998.0                 31.377862               268.0   
4               80998.0                 24.797347               269.0   
...                 ...                       ...                 ...   
99995           50238.0                 34.663572               378.0   
99996           50238.0                 40.565631               379.0   
99997           50238.0                 41.255522               380.0   
99998           50238.0                 33.638208               381.0   
99999           50238.0                 34.192463               382.0   

      Payment_of_Min_Amount  Total_EMI_per_month  Amount_invested_monthly  \
0                        No            49.574949             8.041530e+15   
1                        No            49.574949             1.182802e+16   
2                        No            49.574949             8.169952e+13   
3                        No            49.574949             1.994581e+15   
4                        No            49.574949             4.142015e+16   
...                     ...                  ...                      ...   
99995                    No            35.104023             6.097133e+15   
99996                    No           

In [ ]:
df2

In [ ]:
results

In [ ]:
accuracy

FICO Model
pay_his = ['Delay_from_due_date','Num_of_Delayed_Payment'] # Payment History 35%
debt = ['Outstanding_Debt'] # Amount Debt 30%
cred_his = ['Credit_History_Age'] # Credit History 10%
cred_req = ['Num_Credit_Inquiries'] # New Credit Requests 15%
typ_cred = ['Credit_Mix'] # Type of Credits 10%

Plus 
rate = ['Interest_Rate'] # Interest Rate 5%
cred_ut = ['Credit_Utilization_Ratio'] # Credit Utilization 5%